runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -4328,46 +4328,45 @@ void amReqFn_msgOrdFence(c_nodeid_t node,
     break;
   }
 
+  uint64_t flags = 0;
+  void *ctx = NULL;
+  atomic_bool txnDone;
+
   if (havePutsOut || haveAmosOut) {
-    //
-    // Special case: Do a fenced send if we need it for ordering with
-    // respect to some prior operation(s).  If we can inject, do so and
-    // just collect the completion later.  Otherwise, wait for it here.
-    //
-    if (!blocking
-        && reqSize <= ofi_info->tx_attr->inject_size
-        && envInjectAM) {
-      void* ctx = txnTrkEncodeId(__LINE__);
-      uint64_t flags = FI_FENCE | FI_INJECT;
-      (void) wrap_fi_sendmsg(node, req, reqSize, mrDesc, ctx, flags, tcip);
-    } else {
-      atomic_bool txnDone;
-      void *ctx = txCtxInit(tcip, __LINE__, &txnDone);
-      uint64_t flags = FI_FENCE;
-      (void) wrap_fi_sendmsg(node, req, reqSize, mrDesc, ctx, flags, tcip);
-      waitForTxnComplete(tcip, ctx);
-      txCtxCleanup(ctx);
-    }
+    // Do a fenced send if we need it for ordering with respect to some
+    // prior operation(s).
+    flags |= FI_FENCE;
+  }
 
-    if (havePutsOut) {
-      bitmapClear(tcip->putVisBitmap, node);
-    }
-    if (haveAmosOut) {
-      bitmapClear(tcip->amoVisBitmap, node);
-    }
+  if (reqSize <= ofi_info->tx_attr->inject_size && envInjectAM) {
```

Comment:
The old code didn't bother injecting if this was a blocking AM, on the assumption that if we were going to block anyway there was no point using up provider resources, such as memory buffers or tracking tokens, that might be needed with injection. The new code skips this check. If that's intentional I'm okay with it, but let's note in the PR description why we think it doesn't matter.  And if it's an unintentional oversight then of course let's fix it.